src/ui-test/tests/dev_mode_test.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as consts from '../utils/uiTestConstants';
+import { EditorView, SideBarView, VSBrowser, WebDriver } from 'vscode-extension-tester';
+import { devModeIntegration } from '../../IntegrationConstants';
+import {
+	cleanOutputView,
+	outputViewHasText,
+	viewHasItem,
+	updateFileText,
+	contextMenuItemClick
+} from '../utils/waitConditions';
+import {
+	textDoesNotContainAsci,
+	inputBoxQuickPickOrSet,
+	findSectionItem,
+	DoNextTest
+} from '../utils/utils';
+import { assert } from 'chai';
+
+export function devModeTest(extension: string, language: string, doNextTest: DoNextTest) {
+
+	describe(`Dev Mode test with the integration file update`, function () {
+		this.timeout(60000);
+
+		let driver: WebDriver;
+
+		const [initialCodeMessage, updatedCodeMessage, initialLogMessage, updatedLogMessage] = consts.prepareCodeLogMessages(extension, language);
+
+		before(async function () {
+			driver = VSBrowser.instance.driver;
+		});
+
+		after(async function () {
+			this.timeout(60000);
+			await new EditorView().closeAllEditors();
+			const item = await findSectionItem(consts.extensionName, consts.integrationFileName.toLowerCase());
+			if (item !== undefined) {
+				await driver.wait(() => { return contextMenuItemClick(item, consts.integrationRemove); });
+				const content = new SideBarView().getContent();
+				await driver.wait(async () => {
+					return viewHasItem(content, consts.extensionName, consts.integrationFileName.toLowerCase())
+						.then(val => {
+							return !val
+						});
```

Comment:
I think it is be better to use await notation instead of .then tobe coherent with the other parts of the codebase

Reply:
Copied from last discussion: `return !(await viewHasItem(content, consts.extensionName, consts.integrationFileName.toLowerCase()));`

Before I used `.then` I had tried something like suggested (mb I made some mistake) will try one more, thanks!

Reply:
Done